### PR TITLE
Added Leaky ReLU Activation Function

### PR DESCRIPTION
**Pull request by kausthub-kannan**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: neural_network/activation_functions/leaky_rectified_linear_unit.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        Formula: f(x) = x if x > 0 else f(x) = alpha * x
+
+    Examples:
+    >>> leaky_rectified_linear_unit(vector=np.array([2.3,0.6,-2,-3.8]), alpha=0.3)
+    array([ 2.3 ,  0.6 , -0.6 , -1.14])
+
+    >>> leaky_rectified_linear_unit(vector=np.array([-9.2,-0.3,0.45,-4.56]), \
+    alpha=0.067)
+    array([-0.6164 , -0.0201 ,  0.45   , -0.30552])
+
```
Comment (tianyizheng02) [inline neural_network/activation_functions/leaky_rectified_linear_unit.py:39]: Formatting
```suggestion
    >>> leaky_rectified_linear_unit(np.array([2.3, 0.6, -2, -3.8]), alpha=0.3)
    array([ 2.3 ,  0.6 , -0.6 , -1.14])

    >>> leaky_rectified_linear_unit(np.array([-9.2, -0.3, 0.45, -4.56]), alpha=0.067)
    array([-0.6164 , -0.0201 ,  0.45   , -0.30552])
```

File: neural_network/activation_functions/leaky_rectified_linear_unit.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"""
+Leaky Rectified Linear Unit (LeakyReLU)
+
+Input: vector (type: np.ndarray) , alpha (type: float)
+Output: vector (type: np.ndarray)
+
+UseCase: LeakyReLU solves the issue of dead neurons or vanishing gradient problem.
+Refer the below link for more information:
+https://en.wikipedia.org/wiki/Rectifier_(neural_networks)#Leaky_ReLU
```
Comment (tianyizheng02) [inline neural_network/activation_functions/leaky_rectified_linear_unit.py:9]: Minor grammar/spelling fixes
```suggestion
Leaky Rectified Linear Unit (Leaky ReLU)

Use Case: Leaky ReLU solves the issue of dead neurons, or the vanishing gradient problem.
Refer the below link for more information:
https://en.wikipedia.org/wiki/Rectifier_(neural_networks)#Leaky_ReLU
```

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Just some minor formatting changes, but otherwise LGTM

**Comment by kausthub-kannan** [comment]
Thank you for correcting me. I have done the requested changes with a few improvements.

**Review by tianyizheng02 — APPROVED** [review]

**Comment by tianyizheng02** [comment]
Thanks for your contribution!

**Pull request merged**